### PR TITLE
[Doc] Add contribution difficulty warning

### PR DESCRIPTION
**Pull request by elliotcmorris**
Add some text to the contributing section of the main readme, explaining how OpenAssetIO is a challenging project. Otherwise, it can seem unfair to contributors who might be under the impression that we're a regular project with a big backlog available for work.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,5 +181,22 @@ For detailed instructions, see [building](doc/BUILDING.md).
 - See the [contribution guide](doc/contributing/PROCESS.md)
 - Join our [working group](https://github.com/OpenAssetIO/OpenAssetIO-WG) meetings
 
+> **Warning**
+> OpenAssetIO is a project used in commercial workflows, and due to this
```
Comment (foundrytom) [inline README.md:185]: [consider]
```suggestion
> OpenAssetIO is a project used in business-critical workflows, and due to this
```

File: README.md [hunk 2]
```diff
@@ -181,5 +181,22 @@ For detailed instructions, see [building](doc/BUILDING.md).
...
+> need to work stably and reliably, it is a project with rigid
+> standards.
+>
+> Combined with its inherently abstract nature, it is not a good project
+> to dip your toes lightly into. Beyond minor documentation tweaks, it
+> is difficult to find work appropriate for less experienced
+> implementors.
```
Comment (foundrytom) [inline README.md:192]: [fix] Be good to keep it away from directly negative phrasing, how about:
```suggestion
> Combined with its inherently abstract nature, it is a difficult project
> for casual contribution. We are a small team, and welcome anyone
> who can directly help contribute to the core tasks outlined in our 
> [roadmap](ROADMAP.md). We are however, at present, unable to
> provide "good first tasks" or a gentle, guided, on-ramp for those new 
> to the problem space.
```

File: README.md [hunk 3]
```diff
@@ -181,5 +181,22 @@ For detailed instructions, see [building](doc/BUILDING.md).
...
+>
+> If you have a prevailing interest in the project, and especially if
+> you have VFX industry experience, we are delighted to meet you, but
+> otherwise you may want to consider alternate projects.
+>
+> When planning contributions, you are encouraged to reach out to
+> maintainers directly, as out-of-the-blue PR's are unlikely to succeed.
```
Comment (foundrytom) [inline README.md:199]: [fix] Maybe a little less "go away"?

```suggestion
> If you have a prevailing interest in the project, please reach our and start a 
> thread in the Discussions section. This helps us coordinate work and 
> the approach to ensure we maintain the standard our users demand of the 
> project. Aside from to help fix our glorious range documentation typos, 
> out-of-the-blue PR's without coordination are unlikely to be accepted.

**Review by foundrytom — CHANGES_REQUESTED** [review]
Thanks Elliot - its a subtle one! Attempted to perhaps make it a little more friendly? What do you think?

File: README.md (hunk 1 above)
Comment (elliotcmorris) [reply README.md]: https://github.com/OpenAssetIO/OpenAssetIO/pull/1188/commits/7bc9adfd3e87fb46c991f504dcd671b8fe5f2658

File: README.md (hunk 2 above)
Comment (elliotcmorris) [reply README.md]: https://github.com/OpenAssetIO/OpenAssetIO/pull/1188/commits/81fd7898743250ff2645bd577f7a3a8af743b644

File: README.md (hunk 3 above)
Comment (elliotcmorris) [reply README.md]: I'll admit that was sort of what I was going for, but I see the point.

Edited your suggestion very slightly to remove the whimsy, this is a serious section for serious people doing serious things seriously. I also wanted to leave the door open for obvious non-doc things too, like updating error strings or something like that.

https://github.com/OpenAssetIO/OpenAssetIO/pull/1188/commits/a95f283d3cfe6597ccce817c0473339e75558e08

Comment (foundrytom) [reply README.md]: Noice. Thanks 🙏 sorry for futzing.

**Review by foundrytom — APPROVED** [review]
Thanks @elliotcmorris - wouldn't mind running this by someone not us before we merge if thats OK?

File: README.md [hunk 4]
```diff
@@ -181,5 +181,24 @@ For detailed instructions, see [building](doc/BUILDING.md).
...
+> [roadmap](ROADMAP.md). We are however, at present, unable to
+> provide "good first tasks" or a gentle, guided, on-ramp for those new
+> to the problem space.
+>
+> If you have a prevailing interest in the project, please reach out and
+> start a thread in the Discussions section. This helps us coordinate
+> work and approach to ensure we maintain the standards our users demand
+> of the project. Aside from obvious technical fixes or minor
+> documentation updates, out-of-the-blue PR's without coordination are
+> unlikely to be accepted.
```
Comment (foundrytom) [inline README.md:201]: [fix] Please may we heed JCs words and drop the bit about PRs and have something more along the lines of just "We encourage you to reach out before doing big changes or implementing new features, etc etc”"

Comment (elliotcmorris) [reply README.md]: [92ab8b8](https://github.com/OpenAssetIO/OpenAssetIO/pull/1188/commits/92ab8b8de80305ceccd29327f312daebc378af0c)

**Pull request merged**